### ISSUE TITLE
[Suggestion] Opening Up Binding Options

### ISSUE THREAD

**Issue by PhoxFyre** [coelckers/prboom-plus#193]
As I tend to occasionally use keyboard-only, I am used to having a setup that naturally lets me swap between my keyboard-only and mouse+keyboard layout, but PrBoom+ does have limits as to which buttons can be assigned where, for instance and to my annoyance, the next/previous weapon swaps can only be set to one bind (which for me is either mwforward/mwback or b/v). Is there any way that the keybind capabilities could be expanded to incorporate more than one bind on all binding options?

**Comment by fabiangreffrath** [comment]
This is because in PrBoom+ turning the mouse wheel is translated into a key press event. However, currently I don't see a trivial way out of this, because of the way SDL2 handles the mouse wheel. We could map its movement to mouse buttons but then we would block two of them likewise.

As an aside, in order to allow for both key and mouse button bindings, we would have to maintain two config variables for each action - one for the key and one for the mouse button. That'd be quite some unpleasant work to implement, but given the requests to at least provide this for weapon selection in #129 , this would probably be a good starting point.

**Comment by kraflab** [comment]
In the meantime you can specify a config file with a command line option and keep 2 config files.

**Comment by fabiangreffrath** [comment]
@PhoxFyre I could add actual mouse bindings to the prev/next weapon action, so it would be possible to assign it to the mouse wheel (which gets translated into key press events internally) and to actual mouse buttons at the same time. Would that help?

**Comment by PhoxFyre** [comment]
So it would be similar to how fire can have both a keyboard binding and a mouse binding?

**Comment by fabiangreffrath** [comment]
Yes. But it would still not work around the fact that moving the mouse wheel will be interpreted as pressing a keyboard button. you could bind additional, actual mouse buttons, though.

**Comment by PhoxFyre** [comment]
So effectively it would bind the mouse wheel to keyboard, so then the v button I use to get the last weapon and the b button to get the next weapon would not work along with scrolling forward to get the next weapon and scrolling back to get the last weapon? If there is something I'm confusing I'm sorry. I do not use mouse buttons to change the weapons as I do not own a mouse with more than 3 buttons. But this little focus of changing weapons is a rather trivial complaint and rebinding to either scroll forward/scroll back or b/v wouldn't be too hard to do honestly. I just thought there was a way to possibly expand the binding options to be more similar to GZDoom where its binding options are more free but the backeod of SDL seems to be more strict in its interpreting.

**Comment by fabiangreffrath** [comment]
> If there is something I'm confusing I'm sorry. 

No, it's exactly like that.

**Comment by PhoxFyre** [comment]
> > If there is something I'm confusing I'm sorry.
> 
> No, it's exactly like that.

So both would still be unable to be bound (mwforward/mwbackward & b/v) at the same time?

**Comment by fabiangreffrath** [comment]
> So both would still be unable to be bound (mwforward/mwbackward & b/v) at the same time?

Yes, unfortunately.

> Thanks for the suggestion and possible workaround ho my problem though! Very much appreciated.

Sorry this doesn't really help you, though.
